File: AndroidStudy/dagger2/src/main/java/com/android/dagger2/MyApplication.java
package com.android.dagger2;

import android.app.Application;

import com.android.dagger2.di.ApplicationComponent;
import com.android.dagger2.di.DaggerApplicationComponent;
import com.android.dagger2.di.NetworkModule;

public class MyApplication extends Application {
    static ApplicationComponent applicationComponent;

    @Override
    public void onCreate() {
        super.onCreate();
        applicationComponent = DaggerApplicationComponent.builder().networkModule(new NetworkModule(this)).build();
    }

    public static ApplicationComponent getApplicationComponent() {
        return applicationComponent;
    }
}

File: JetpackStudy/lifecycle/src/main/java/com/android/lifecycle/MyApplication.kt
package com.android.lifecycle

import android.app.Application
import androidx.lifecycle.ProcessLifecycleOwner

class MyApplication : Application() {
    override fun onCreate() {
        super.onCreate()
        //针对整个应用程序的监听，与Activity数量无关
        //Lifecycle.Event.ON_CREATE只会被调用一次，Lifecycle.Event.ON_DESTROY永远不会调用
        ProcessLifecycleOwner.get().getLifecycle().addObserver(ApplicationObserver())
    }
}